### PR TITLE
Remove Ember.String.fmt

### PR DESCRIPTION
**Pull request by locks**

### CODE REVIEW

**Comment by rwjblue** [comment]
Seems good to me.  We need to ensure this functionality sticks around when you have `ember-2-legacy` addon installed, but its super easy to just move the implementation over there. 

Can you make an issue over on https://github.com/emberjs/ember-2-legacy to add support to add these back when the legacy addon is present?  I just don't want to lose track...

File: packages/ember-runtime/lib/system/string.js [hunk 1]
```diff
@@ -49,35 +49,6 @@ const STRING_DECAMELIZE_REGEXP = (/([a-z\d])([A-Z])/g);
 
 const DECAMELIZE_CACHE = new Cache(1000, str => str.replace(STRING_DECAMELIZE_REGEXP, '$1_$2').toLowerCase());
 
-function _fmt(str, formats) {
```
Comment (sduquej) [inline packages/ember-runtime/lib/system/string.js]: `_fmt` is used in `loc`:

https://github.com/emberjs/ember.js/blob/a7518b940908994ae0577973987fd4cb6a5f7bee/packages/ember-runtime/lib/system/string.js#L81-L88

and also referenced in the [documentation for it](https://emberjs.com/api/ember/2.17/classes/String/methods/loc?anchor=loc).

Think we'll need to update these too.

**Comment by lorcan** [comment]
`Ember.String.loc` also depends on the same functionality of `Ember.String.fmt`. Is there a case for deprecating `Ember.String.loc`? Too late for 3.0.0, but worth flagging?

**Comment by locks** [comment]
@lorcan thanks for the review. `loc` is already deprecated by [RFC 236](https://github.com/emberjs/rfcs/blob/master/text/0236-deprecation-ember-string.md), and you can follow the implementation of this deprecation in https://github.com/emberjs/ember.js/issues/15624.
As mentioned in [The Road to Ember 3.0](https://emberjs.com/blog/2017/10/03/the-road-to-ember-3-0.html), "No additional 3.0-targeted deprecations will be introduced."

If we manage to finish all the `Ember.String` work soon, the deprecation should land in the 3.x series, with an until of 4.0.0.

**Review by rwjblue — APPROVED** [review]

**Pull request merged**